### DISCUSSION TITLE
Magick.NET and creating a hidden text watermark

### DISCUSSION THREAD

**Discussion by cblaze22** [dlemstra/Magick.NET#1527] [Help]
Any help on this question would be great.

https://stackoverflow.com/questions/77875340/magick-net-and-creating-a-hidden-text-watermark

*Maintainer edit (added post content from stack overflow):*

Is there a way to create a hidden watermark text so we can verify if our images were copied on another site in the future? We are currently playing around with this code below but it just writes the text all over the image, instead of it being hidden and not seen. Is there anything other then a watermark I can do to handle this situation with Magick.NET?

The below code doesn't seem to be getting 0 back on new file.

```C#
 using (var message = new MagickImage("label:© Events", 200, 20))
 {
     using (var image = new MagickImage(imagePath))
     {
         image.Stegano(message);
         image.Write(imageSteganoPath);

         var settings = new MagickReadSettings();
         settings.Format = MagickFormat.Stegano;
         settings.Width = 200;
         settings.Height = 20;

         using (var hiddenMessage = new MagickImage(imageSteganoPath, settings))
         {
             Assert.AreEqual(0, message.Compare(hiddenMessage, ErrorMetric.RootMeanSquared));
         }
     }
 }
```

**Comment by dlemstra** [comment]
I don't understand what you are trying to do.

**Reply by cblaze22** [reply]
I’m trying to write a hidden message into the image to verify if someone is using it. Using Stegano the reversing of the label doesn’t show 0 to verify the label. Not sure I’m doing it right but this code was taken from a past Magick example.

**Reply by dlemstra** [reply]
Your stack overflow post says this:

> We are currently playing around with this code below but it just writes the text all over the image, instead of it being hidden and not seen.

I don't see the text when I use the `Stegano` method?

> Using Stegano the reversing of the label doesn’t show 0 to verify the label. 

The value will be close to zero? It might be a tiny bit higher than zero.

**Comment by cblaze22** [comment]
It’s like 2.4 or something. I’ll test it without and see how large it is
with out it. I figured it would have been 0 though thanks.

Sorry some of that was old comments, with all over the image.

Mike
